tsconfig.json

```diff
@@ -3,8 +3,8 @@
 		"declaration": true,
 		"declarationMap": true,
 		"esModuleInterop": true,
-		"module": "ESNext",
-		"moduleResolution": "Bundler",
+		"module": "nodenext",
+		"moduleResolution": "nodenext",
```

Comment:
Using this for `moduleResolution` is sufficient to enforce explicit extensions, without an additional eslint rule.  `Bundler` is intended for packages that actually bundle their code, which is why we kept getting tripped up by this.

Reply:
Oh nice. Do you think this should be upstreamed to https://github.com/JoshuaKGoldberg/create-typescript-app then?

Reply:
Yeah, if CTA is prioritizing not bundling, then nodenext would be the better option.

Reply:
Oh - I just checked and it's already using nodenext. https://github.com/JoshuaKGoldberg/create-typescript-app/blob/afc398b0f27c6b6e8699e7e0d471c64438bbac04/src/blocks/blockTypeScript.ts#L155-L156. Great.